### PR TITLE
Add changelog

### LINKED ISSUE #292: [Feature]: Add a changelog to the repo
### What is the feature you would like to see?

Introduce changelog and a way to show difference between releases

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by ggonzalez94**
Resolves #292 

Sets the general structure for a Changelog that we can follow starting with the changes for `v0.2.0`. 
Structure and how to maintain it based on: https://keepachangelog.com/en/1.1.0/ 

DO NOT MERGE UNTIL WE RELEASE `0.1.0` SINCE WE ARE SPECIFYING THE DATE OF THAT RELEASE

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+-
+
+### Fixed
+
+-
+
+## [0.1.0] - 2024-10-16
+
+- Initial release
```
Comment (bidzyyys) [inline CHANGELOG.md:24]: Should we add here everything?

Comment (ggonzalez94) [reply CHANGELOG.md:24]: We could... but not sure if it's worth it. I would focus more on being disciplined from this point onwards. I remember finding this online "While you don't need to provide detailed information about the first release, it's good to acknowledge it" and that's the approach I followed

**Comment by ggonzalez94** [comment]
@qalisander the step that sets up the nitro testnode is constantly failing(I've re run multiple times) even if the changes don't touch any of the code. Do you have an idea of why this is?

**Review by bidzyyys — APPROVED** [review]
LGTM

**Pull request merged**